sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -62,6 +63,14 @@ trait InvokeLike extends Expression with NonSQLExpression with ImplicitCastInput
       .map(cls => v => cls.cast(v))
       .getOrElse(identity)
 
+  // Returns true if we can trust all values of the given DataType can be serialized.
+  private def trustedSerializable(dt: DataType): Boolean = {
+    // Right now we conservatively block all ObjectType (Java objects) regardless of
+    // serializability, because the type-level info with java.io.Serializable and
+    // java.io.Externalizable marker interfaces are not strong guarantees.
+    // This restriction can be relaxed in the future to expose more optimizations.
+    !dt.isInstanceOf[ObjectType]
```

Comment:
to be safe, how about `!dt.existsRecursively(_.isInstanceOf[ObjectType])`?

Reply:
> to be safe, how about `!dt.existsRecursively(_.isInstanceOf[ObjectType])`?

+1

Reply:
Thank you both very much! This is a great suggestion, addressed in latest commit.